imagetest/test_suites/storageperf/storage_perf_utils.go

```diff
@@ -207,3 +214,113 @@ func checkZypperTransientError(err error, stdout, stderr string) error {
 	}
 	return err
 }
+
+// use the guest attribute to check if hyperdisk is being used. If the guest attribute was not set, assume by default that hyperdisk fio options are not being used.
+func isUsingHyperdisk(ctx context.Context) bool {
+	diskType, err := utils.GetMetadata(ctx, "instance", "attributes", diskTypeAttribute)
+	if err != nil {
+		return false
+	}
+	if diskType == imagetest.HyperdiskExtreme || diskType == imagetest.HyperdiskThroughput || diskType == imagetest.HyperdiskBalanced {
+		return true
+	}
+
+	return false
+}
+
+// function to get num numa nodes
+func getNumNumaNodes() (int, error) {
```

Comment:
This won't work on windows

Reply:
This is a large PR, so I'll plan on a follow up PR for windows hyperdisk